tests/gradient_z_test.cpp

```diff
@@ -0,0 +1,237 @@
+
+
+#include "clesperanto.hpp"
+
+#include <random>
+
+template <class type>
+auto
+run_test(const std::array<size_t, 3> & shape, const cle::MemoryType & mem_type) -> bool
+{
+  std::vector<type> input = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+  std::vector<type> valid = { 0, 0, 0, 0, -1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0 };
```

Comment:
The clij OpenCL convention for the gradient is (1, -1) order, Not (-1, 1) order.
Invert the -1 and the 1 at line 12 and you should be good